File: backend/src/main/kotlin/bo/edu/umsa/backend/dto/ProjectDashboardDto.kt
package bo.edu.umsa.backend.dto

data class ProjectDashboardDto(
    val totalProjects: Int,
    val totalCompletedProjects: Int,
    val totalCompletedWithDelayProjects: Int,
    val totalInProgressProjects: Int,
    val totalDelayedProjects: Int,
    val projectByDate: List<ProjectByDateDto>,
)

data class ProjectByDateDto(
    val year: Int,
    val month: Int,
    val completedProjects: Int,
    val completedWithDelayProjects: Int,
    val inProgressProjects: Int,
    val delayedProjects: Int,
)

File: frontend/src/app/features/home/models/project-dashboard.dto.ts
export interface ProjectDashboardDto {
    totalProjects: number;
    totalCompletedProjects: number;
    totalCompletedWithDelayProjects: number;
    totalInProgressProjects: number;
    totalDelayedProjects: number;
    projectByDate: ProjectByDateDto[];
}

export interface ProjectByDateDto {
    year: number;
    month: number;
    completedProjects: number;
    completedWithDelayProjects: number;
    inProgressProjects: number;
    delayedProjects: number;
}
